src/GitHub.Api/Installer/GitInstaller.cs

```diff
@@ -129,9 +130,48 @@ public void SetupGitIfNeeded(ActionTask<NPath> onSuccess, ITask onFailure)
         private void ExtractPortableGit(ActionTask<NPath> onSuccess, ITask onFailure)
         {
             ITask downloadFilesTask = null;
-            if ((gitArchiveFilePath == null) || (gitLfsArchivePath == null))
+
+            if (gitArchiveFilePath == null || gitLfsArchivePath == null)
             {
-                downloadFilesTask = CreateDownloadTask();
+                archiveParentPath = NPath.CreateTempDirectory("git_zip_paths");
+
+                if (gitArchiveFilePath == null)
+                {
+                    gitArchiveFilePath = AssemblyResources.ToFile(ResourceType.Platform, "git.zip", archiveParentPath, environment);
+                    if (!gitArchiveFilePath.FileExists())
+                    {
+                        gitArchiveFilePath = null;
```

Comment:
Should this file exists check be moved into `AssemblyResources.ToFile(...)`?

Reply:
I also was not sure if the usage of `AssemblyResources.ToFile(...)` be used here inside `GitInstaller` or the result sent into the constructor.

Reply:
> I also was not sure if the usage of AssemblyResources.ToFile(...) be used here inside GitInstaller or the result sent into the constructor.

Now that I say that out loud I feel the like call should be outside `GitInstaller`...

Reply:
Yeah, I agree it probably should be outside, this should only be concerned with extracting a zip given a file path.